app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -45,21 +49,31 @@ class ProfileForm extends Component {
   }
 
   handleChange(e) {
+    const { formValues } = { ...this.state };
+    const currentFormState = formValues;
     const { name, value } = e.target;
 
-    this.setState({
-      [name]: value,
-    });
+    currentFormState[name] = value;
+
+    // Once we've derived the new form values, check if the form is empty
+    // and use that value to set the `canSkip` property on the state.
+    const formIsEmpty =
+      Object.values(currentFormState).filter((v) => v.length > 0).length === 0;
```

Comment:
I don't love this, but it works! I am open to other ways to write this :)

Reply:
I like it 😆 . I can't think of anything better